numpy/fft/fftpack.py

```diff
@@ -86,7 +92,8 @@ def _raw_fft(a, n=None, axis=-1, init_function=fftpack.cffti,
     # As soon as we put wsave back into the cache, another thread could pick it
     # up and start using it, so we must not do this until after we're
     # completely done using it ourselves.
-    fft_cache[n].append(wsave)
+    with _cache_lock:
+        fft_cache[n].append(wsave)
```

Comment:
Do you mean the other snippet where the cache is accessed? There are many ways of achieving the same thing but I'm not sure any results in clearer logic. Can you write a short snippet of what you are thinking of?